user-impl/src/main/java/com/example/auction/user/impl/UserRepository.java

```diff
@@ -79,6 +81,19 @@ public UserRepository(CassandraSession session, ReadSide readSide) {
                 .thenApply(TreePVector::from);
     }
 
+    protected CompletionStage<UUID> getUserIdByEmail(String email) {
+
+        return session
+            .selectOne(
+                "SELECT * FROM UserInfo  WHERE email = ? " ,
+                   email
+            )
+
+            .thenApply(rows -> rows.map(UserRepository::convertUserSummary).get())
+            .thenApply(UserSummary -> UserSummary.getId());
+
```

Comment:
Good visual formatting. Each operation on a different line helps quick scanning of code and understanding what it's doing. 👍